frame/fee-market/src/lib.rs

```diff
@@ -241,53 +245,73 @@ pub mod pallet {
 				T::RingCurrency::free_balance(&who) >= new_collateral,
 				<Error<T>>::InsufficientBalance
 			);
-			ensure!(
-				new_collateral > Self::get_relayer(&who).collateral,
-				<Error<T>>::OnlyIncreaseLockedCollateralAllowed
-			);
-
-			Self::update_collateral(&who, new_collateral);
-			Self::deposit_event(Event::<T>::UpdateLockedCollateral(who, new_collateral));
-			Ok(().into())
-		}
 
-		/// Cancel enrolled relayer
-		#[pallet::weight(<T as Config>::WeightInfo::cancel_enrollment())]
-		#[transactional]
-		pub fn cancel_enrollment(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin)?;
-			ensure!(Self::is_enrolled(&who), <Error<T>>::NotEnrolled);
+			let relayer = Self::relayer(&who);
 			ensure!(
-				<Relayers<T>>::get().len() > T::AssignedRelayersNumber::get() as usize,
-				<Error<T>>::TooFewEnrolledRelayers
+				new_collateral > relayer.collateral,
+				<Error<T>>::OnlyIncCollateralAllowed
 			);
-			ensure!(!Self::is_occupied(&who), <Error<T>>::OccupiedRelayer);
+			let old_capacity: u32 =
+				(relayer.collateral / T::CollateralPerOrder::get()).saturated_into::<u32>();
+			let used_capacity = old_capacity.saturating_sub(relayer.order_capacity);
+			let new_capacity: u32 =
+				(new_collateral / T::CollateralPerOrder::get()).saturated_into::<u32>();
```

Comment:
cache updating, can be removed if we replace cache with calculation function.

Reply:
See the comments above.